### DISCUSSION TITLE
ChildFetch in class with object factory

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#1654] [Questions]
I am having trouble invoking a [CreateChild] CreateChild() when using the ObjectFactory redirect at the top of the class.  I thought only the create, get, update, and deletes where redirected to the factory.  Here's an example of a method I cannot invoke from my view layer.  
     
```c#
[Serializable]
        [Csla.Server.ObjectFactory(typeof(SupplierFactory), "Create", "GetAsync", "UpdateAsync", "DeleteAsync")]
        Public Class Supplier : BusinessBase <SupplierInfo>
        {
         [CreateChild]
        protected void CreateChild()
        {
            var result = DataPortal.Create<SupplierInfoPersist>();

            string[] args = { "OriginalValues" };

            using (BypassPropertyChecks)
            {
                DataMapper.Map(result, this, args);
                MarkAsChild();
                MarkNew();
            }
            }
        }
```
 

Can anyone let me know what I need to do instead?

**Comment by rockfordlhotka** [comment]
If I recall correctly, if the `ObjectFactory` attribute is applied then _all_ data portal operation methods are redirected to the factory type.

The method name parameters you specify are optional, and control the names of the methods in the factory, otherwise they revert to the defaults and/or the ones designated by the modern attributes.

**Comment by kcabral817** [comment]
okay.  I got it!  I will see it in the morning....well a few hours from now :-)  Thank you.
